cpp/ql/src/semmle/code/cpp/internal/ResolveClass.qll

```diff
@@ -1,9 +1,9 @@
 import semmle.code.cpp.Type
 
 pragma[noinline]
-private string getTopLevelClassName(@usertype c) {
+private string getTopLevelClassMangledName(@usertype c) {
   isClass(c) and
-  usertypes(c, result, _) and
+  mangled_type_name(c, result) and
```

Comment:
There needs to be either a fallback for when `mangled_type_name` is empty, or there needs to be a complete upgrade script.